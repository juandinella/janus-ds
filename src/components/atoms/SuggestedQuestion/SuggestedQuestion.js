import isPropValid from '@emotion/is-prop-valid'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import styles from './SuggestedQuestion.styles'

/**
 * SuggestedQuestion
 */

const shouldForwardProp = (prop) => isPropValid(prop)
const StyledSuggestedQuestion = styled.button.withConfig({ shouldForwardProp })`
  ${styles}
`

const SuggestedQuestion = ({ id, children, onClick = () => null }) => {
  return (
    <StyledSuggestedQuestion id={id} data-testid={id} onClick={onClick || null}>
      <span>{children}</span>
    </StyledSuggestedQuestion>
  )
}

SuggestedQuestion.displayName = 'SuggestedQuestion'

SuggestedQuestion.propTypes = {
  id: PropTypes.string,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
}

export default SuggestedQuestion
