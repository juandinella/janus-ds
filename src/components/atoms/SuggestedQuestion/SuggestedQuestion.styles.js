import { css } from 'styled-components'

export default css`
  display: flex;
  padding: var(--spacing-xxs) var(--spacing-xs);
  background-color: var(--color-neutral-100);
  border-radius: var(--border-radius-sm);
  font-size: var(--font-size-sm);
  transition: background-color var(--transition-225);
  padding: var(--spacing-xs);

  &:hover {
    background-color: var(--color-neutral-200);
  }
`
