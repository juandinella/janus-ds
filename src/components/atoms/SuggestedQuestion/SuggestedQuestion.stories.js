import SuggestedQuestion from './SuggestedQuestion'

import { getTemplate } from '../../../helpers/storybook'

const Template = getTemplate(SuggestedQuestion)

export default {
  title: 'Atoms/SuggestedQuestion',
  component: SuggestedQuestion,
  args: {
    children: 'What is Janus?',
  },
  argTypes: {},
}

export const Default = Template.bind({})
