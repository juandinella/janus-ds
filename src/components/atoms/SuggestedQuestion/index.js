import SuggestedQuestion from './SuggestedQuestion'

export default SuggestedQuestion
