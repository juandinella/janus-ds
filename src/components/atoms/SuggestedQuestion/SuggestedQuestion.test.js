import initStoryshots from '@storybook/addon-storyshots'
import { render, screen, cleanup, fireEvent } from '@testing-library/react'
import React from 'react'
import SuggestedQuestion from './SuggestedQuestion'

initStoryshots({
  storyKindRegex: /^Atoms\/SuggestedQuestion$/,
})

const componentProps = {
  id: 'unique_id',
  onClick: jest.fn(),
}
afterEach(cleanup)
beforeEach(() => {
  componentProps.onClick.mockClear()
})

describe('<SuggestedQuestion />', () => {
  test('should render component correctly', () => {
    render(<SuggestedQuestion {...componentProps} />)
    const btn = screen.getByTestId(componentProps.id)
    expect(btn).toBeInTheDocument()
  })
  test('should call function on click', () => {
    const { getByTestId } = render(<SuggestedQuestion {...componentProps} />)
    const button = getByTestId(componentProps.id)
    fireEvent.click(button)
    fireEvent.click(button)
    expect(componentProps.onClick).toHaveBeenCalledTimes(2)
  })
})
