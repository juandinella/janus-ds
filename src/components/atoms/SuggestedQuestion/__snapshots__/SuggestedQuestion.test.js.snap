// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/SuggestedQuestion Default 1`] = `
<div
  style={
    {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "position": "relative",
    }
  }
>
  <button
    className="sc-18ktvrn-0 kRLHHi"
    onClick={[Function]}
  >
    <span>
      What is Janus?
    </span>
  </button>
</div>
`;
